tfexec/cmd.go

```diff
@@ -193,6 +195,32 @@ func (tf *Terraform) buildTerraformCmd(ctx context.Context, mergeEnv map[string]
 	return cmd
 }
 
+func (tf *Terraform) buildTerraformCmdWithGracefulShutdown(ctx context.Context, gracefulShutdownPeriod time.Duration, mergeEnv map[string]string, args ...string) *exec.Cmd {
+	cmd := exec.CommandContext(ctx, tf.execPath, args...)
+
+	cmd.Env = tf.buildEnv(mergeEnv)
+	cmd.Dir = tf.workingDir
+
+	if gracefulShutdownPeriod > 0 {
+		tf.logger.Printf("[INFO] configuring gracefully cancellation for command: %s", cmd.String())
+		cmd.WaitDelay = gracefulShutdownPeriod
```

Comment:
question: I don't know much about the `cmd.WaitDelay`. I guess when we send `os.Interrupt` to the subprocess, the subprocess must exit in `cmd.WaitDelay`. Does my understanding right?

Reply:
Pretty much, after `cmd.WaitDelay`, the standard library will send kill signal for force quitting the subprocess. This is the corresponding implementation: https://cs.opensource.google/go/go/+/master:src/os/exec/exec.go;l=793-813;drc=a548dee75cc2fd578acc303f58887a50e64e29bc